pkg/daemonsetlock/daemonsetlock.go

```diff
@@ -45,7 +45,9 @@ func (dsl *DaemonSetLock) Acquire(metadata interface{}, TTL time.Duration) (acqu
 			}
 
 			if ttlExpired(value.Created, value.TTL) {
-				return true, value.NodeID, nil
+				if err := dsl.Release(); err != nil {
```

Comment:
It seems suboptimal to do another read-modify-write cycle here: can't you just skip the return on line 51 if the ttl has expired?

Reply:
If I understand https://github.com/weaveworks/kured/blob/1.4.0/pkg/daemonsetlock/daemonsetlock.go#L38-L44 correctly the loop iterates through annotations and if there is one, it returns result of `value.NodeID == dsl.nodeID`.

Not enabling this feature will keep `TTL=0`, which means `ttlExpired` will always return `false` and return properly if needed. Otherwise, it will skip to [this line](https://github.com/weaveworks/kured/blob/ec755333941a4a647ac3d782c29016e89ea540c0/pkg/daemonsetlock/daemonsetlock.go#L64).